libs/core/libs/shared/src/core_ops.rs

```diff
@@ -75,50 +79,50 @@ where
         Ok(File { id, parent, name, file_type, last_modified, last_modified_by, shares })
     }
 
-    // this variant used when we want to skip certain files e.g. when listing paths
-    pub fn resolve_and_finalize<I>(
-        &mut self, account: &Account, ids: I, public_key_cache: &mut LookupTable<Owner, String>,
-    ) -> SharedResult<Vec<File>>
-    where
-        I: Iterator<Item = Uuid>,
-    {
-        let mut user_visible_ids = Vec::new();
-        for id in ids {
-            if self.calculate_deleted(&id)? {
-                continue;
-            }
-            if self.in_pending_share(&id)? {
-                continue;
-            }
-            if self.link(&id)?.is_some() {
-                continue;
+    //finalizes one single id.
+    pub fn finalize(
+        &mut self, account: &Account, id: Uuid, public_key_cache: &mut LookupTable<Owner, String>,
+    ) -> SharedResult<File> {
+        let file;
+
+        let finalized = self.decrypt(&id, account, public_key_cache)?;
+
+        match finalized.file_type {
```

Comment:
...declare/assign `file` here...
```suggestion
        let file = match finalized.file_type {
```